modules/platforms/dotnet/Apache.Ignite.Core/Datastream/IDataStreamer.cs

```diff
@@ -138,24 +134,44 @@ public interface IDataStreamer<TK, TV> : IDisposable
         /// <para />
         /// Setter must be called before any add/remove operation.
         /// <para />
-        /// Default is 0, which means Ignite calculates this automatically as 
-        /// <see cref="IgniteConfiguration.DataStreamerThreadPoolSize"/> * 
+        /// Default is 0, which means Ignite calculates this automatically as
+        /// <see cref="IgniteConfiguration.DataStreamerThreadPoolSize"/> *
         /// <see cref="DataStreamerDefaults.DefaultParallelOperationsMultiplier"/>.
         /// </summary>
         int PerNodeParallelOperations { get; set; }
 
         /// <summary>
-        /// Automatic flush frequency in milliseconds. Essentially, this is the time after which the
-        /// streamer will make an attempt to submit all data added so far to remote nodes.
-        /// Note that there is no guarantee that data will be delivered after this concrete
-        /// attempt (e.g., it can fail when topology is changing), but it won't be lost anyway.
+        /// Gets or sets the automatic flush frequency, in milliseconds.
+        /// Data streamer buffers the data for performance reasons.
+        /// The buffer is flushed in the following cases:
+        /// <ul>
+        /// <li>Buffer is full.</li>
+        /// <li><see cref="Flush"/> or <see cref="TryFlush"/> is called.</li>
+        /// <li>Periodically when <see cref="AutoFlushInterval"/> is set.</li >
+        /// </ul>
         /// <para />
         /// If set to <c>0</c>, automatic flush is disabled.
         /// <para />
         /// Default is <c>0</c> (disabled).
         /// </summary>
+        [Obsolete("Use AutoFlushInterval.")]
```

Comment:
Should not there be a justification for deprecation?

Reply:
The justification is that `TimeSpan` should be used for time interval purposes
* No ambiguity about units
* .NET-native type
* Consistent with other Ignite APIs and system APIs

However, I don't think we should provide those details in the deprecation message.